### PR TITLE
Added support for SplFileInfo for Finder

### PR DESCRIPTION
**Pull request by theofidry**
@teohhanhui as promised here's the PR.

* Added tests for the `Finder` component, resulted in some bugfix (removed the use of Doctrine Loader)
* Added the support for `SplFileInfo` as requested in #64

### CODE REVIEW

**Comment by teohhanhui** [comment]
Is there any reason to limit it to `Symfony\Component\Finder\SplFileInfo` instead of the base `SplFileInfo`?

**Comment by theofidry** [comment]
None, good point.

**Comment by theofidry** [comment]
@teohhanhui confirm me if it's good on your side.

File: Doctrine/Finder/Finder.php [hunk 1]
```diff
@@ -23,37 +23,63 @@
...
-            foreach ($loader->getFixtures() as $_loader) {
-                /** @var Loader $_loader */
-                $fixtures = array_merge($fixtures, $_loader->getFixtures());
-            }
-        } else {
-            $finder->in($path)->files()->name('*.yml');
-            foreach ($finder as $file) {
-                /** @var SplFileInfo $file */
-                $fixtures[] = $file->getRealPath();
+        // Check if PHP classes are data loaders or not
```
Comment (teohhanhui) [inline Doctrine/Finder/Finder.php]:
```php
        // Check if PHP classes are data loaders or not
        foreach (get_declared_classes() as $className) {
            $reflectionClass = new \ReflectionClass($className);
            $sourceFile = $reflectionClass->getFileName();

            if (true === isset($phpClasses[$sourceFile])) {
                if ($reflectionClass->implementsInterface('Hautelook\AliceBundle\Doctrine\DataFixtures\LoaderInterface')) {
                    $loaders[] = new $className;
                }
            }
        }
```

**Pull request merged**